File: Domain/Entities/Appointment.cs
﻿using Domain.Base;

namespace Domain.Entities
{
    public class Appointment : BaseEntity
    {
        public DateTime CheckInDate { get; set; }
        public DateTime CheckOutDate { get; set; }
        public string Status { get; set; }
        public string Reason { get; set; }
        public string Notes { get; set; }
        public Guid DoctorId { get; set; }
        public Doctor Doctor { get; set; }
        public Guid PatientId { get; set; }
        public Patient Patient { get; set; }
    }
}

File: client/src/app/models/Appointment.ts
import Doctor from "./Doctor";
import Patient from "./Patient";

export default interface Appointment {
    id: string;
    createdAt: Date;
    updatedAt: Date;
    checkInDate: Date;
    checkOutDate: Date;
    status: string;
    reason: string;
    notes: string;
    doctorId: string;
    doctor: Doctor;
    patientId: string;
    patient: Patient;

}
